app/questionnaire/questionnaire_schema.py

```diff
@@ -206,9 +206,33 @@ def _get_error_messages(self):
 
         return messages
 
-    def _get_answer_dependencies(self):
+    def _get_answers_by_id_for_question(self, question_id):
+        question = self.get_question(question_id)
+
+        return {
+            answer['id']: answer
+            for answer in question['answers']
+        }
+
+    def _get_answer_ids_for_question(self, question_id):
+        """ get answer ids associated with a specific question_id """
+        return list(self._get_answers_by_id_for_question(question_id).keys())
+
+    def _get_answers_dependencies(self):
         dependencies = defaultdict(list)
+
+        dependencies.update(self._get_answer_level_dependencies())
+
+        dependencies.update(self._get_question_level_dependencies())
+
+        dependencies.update(self._get_titles_dependencies_for_all_questions())
+
```

Comment:
This is a type of question level dependency so rather than loop through the questions type twice it's worth calling `_get_titles_dependencies_for_question(...)` within `_get_question_level_dependencies()`. It may also be worth moving the existing code in